### PR TITLE
PIM-4732: Added a new cookbook about storage events

### PR DESCRIPTION
**Pull request by ClementGautier**

### CODE REVIEW

**Comment by sumbobyboys** [comment]
GTM

File: reference/events/storage_events.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+`CheckPublishedProductOnRemovalSubscriber (Enterprise edition)`_  0
+================================================================  ===============
+
+.. _CheckChannelsOnDeletionSubscriber: https://github.com/akeneo/pim-community-dev/blob/master/src/Pim/Bundle/CatalogBundle/EventSubscriber/Category/CheckChannelsOnDeletionSubscriber.php
+.. _ProductTemplateAttributeSubscriber: https://github.com/akeneo/pim-community-dev/blob/master/src/Pim/Bundle/CatalogBundle/EventSubscriber/ProductTemplateAttributeSubscriber.php
+.. _RuleRelationSubscriber (Enterprise edition): https://github.com/akeneo/pim-enterprise-dev/blob/master/src/PimEnterprise/Bundle/CatalogRuleBundle/EventSubscriber/RuleRelationSubscriber.php
+.. _RemoveChannelSubscriber (Enterprise edition): https://github.com/akeneo/pim-enterprise-dev/blob/master/src/PimEnterprise/Bundle/ProductAssetBundle/EventSubscriber/ORM/RemoveChannelSubscriber.php
+.. _CheckPublishedProductOnRemovalSubscriber (Enterprise edition): https://github.com/akeneo/pim-enterprise-dev/blob/master/src/PimEnterprise/Bundle/WorkflowBundle/EventSubscriber/PublishedProduct/CheckPublishedProductOnRemovalSubscriber.php
```
Comment (nidup) [inline reference/events/storage_events.rst]: enterprise repository has restricted access

Comment (ClementGautier) [inline reference/events/storage_events.rst]: What should I do then ? I did try to remove the link and display the full file path before and it adds an horizontal scroll that kills the user experience.

File: cookbook/event/index.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
+How to implement your business logic using the event system
+===========================================================
+
+The PIM use the Symfony event system and fires events when something happens inside the system. For you, this means that
+you can create your own `services`_ that interact with those  objects whenever certain action happens within the PIM.
```
Comment (nidup) [inline cookbook/event/index.rst]: those  objects => those objects

Comment (ClementGautier) [inline cookbook/event/index.rst]: In see no difference

File: reference/events/storage_events.rst (hunk 1 above)
Comment (nidup) [inline reference/events/storage_events.rst]: yep would say full path or maybe no link at all and just the name of the subscriber for the EE (what i did for the components and bundles pages)

File: cookbook/event/index.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
+How to implement your business logic using the event system
+===========================================================
+
+The PIM use the Symfony event system and fires events when something happens inside the system. For you, this means that
```
Comment (MarieMinasyan) [inline cookbook/event/index.rst]: uses

File: cookbook/event/index.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+you can create your own `services`_ that interact with those objects whenever certain action happens within the PIM.
+
+.. note::
+
+    We highly recommend you to implement your business code on listeners, this way you will not depends on Akeneo or Pim
```
Comment (MarieMinasyan) [inline cookbook/event/index.rst]: recommend you implement... this way you will not depend on

File: cookbook/event/index.rst [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+For more details about event listeners, you can read the `Symfony documentation`_
+
+.. _Symfony documentation: http://symfony.com/doc/current/cookbook/event_dispatcher/event_listener.html
+.. _services: http://symfony.com/doc/current/book/service_container.html
+
+Example
+-------
+
+In the following example we will assume you want to be emailed about each product modification.
```
Comment (MarieMinasyan) [inline cookbook/event/index.rst]: you want to receive an email whenever a product modification occurs

File: reference/events/index.rst [hunk 6]
```diff
@@ -0,0 +1,11 @@
+Events
+======
+
+.. note::
+
+    For a description of event usage, please refere to the :doc:`cookbok </cookbook/event/index>`
```
Comment (MarieMinasyan) [inline reference/events/index.rst]: refer

**Pull request merged**